packages/js/product-editor/src/components/variations-table/use-variations/use-variations.ts

```diff
@@ -371,14 +374,18 @@ export function useVariations( { productId }: UseVariationsProps ) {
 
 			currentPage++;
 
-			result.push( ...( response?.update ?? [] ) );
+			const updatedVariations = response?.update ?? [];
+			result.push( ...updatedVariations );
 
-			for ( const variation of subset ) {
+			for ( const variation of updatedVariations ) {
 				await coreInvalidateResolution( 'getEntityRecord', [
 					'postType',
 					'product_variation',
 					variation.id,
 				] );
+
+				selectedVariationsRef.current[ variation.id ] = variation;
+				setSelectedCount( ( current ) => current + 1 );
```

Comment:
Do we need to call this within this loop?
Couldn't we just call it outside of this loop by using `Object.keys(selectedVariationsRef.current).length`?

Reply:
Well I need to call `setSelectedCount` to produce a change in the `selected` variations to notify the consumers, this call can be made for the whole batch update, but since the UI manage to show partial updated data chunks then it is nicer to notify consumers per chunk. Meaning the notification must happens within the while loop to achieve that. Then it does not matter if I keep the notification within the for loop or move it outside of it, react batches the update and execute it at the right moment. But if you strongly consider this needs to be moved out of the for loop, I can change it to `setSelectedCount( ( current ) => current + updatedVariations.length );` instead.

Reply:
No that's fine then, I just wanted to know the reasoning and double check if it affected the React rendering state.
Your consensus makes sense.